Default sprectral weighting

ryancoe opened the pull request:
## Description
I found the toggling of warnings in the example file particularly offensive for some reason :), so I made changes to set the equal weighting of the example spectra. This relates to #108, although does not close it.

Fixes (none)

[test_results.pdf](https://github.com/SNL-WaterPower/WecOptTool/files/4582231/test_results.pdf)

## Checklist:

- [x] Added the results of running the test suite (in pdf form)
- [x] All new files contain the GPL header
- [x] If `example.m` has been modified, the content / line numbers in 
  `docs\user\example.rst` are still valid or have been fixed

H0R5E commented:
I think this is a good idea, as it looks a bit shonky having an example that triggers our own warning. Before I review this, though, I would like to suggest a couple of things to add value to the PR:

1. Take the opportunity to move this warning outside of the optimiser by moving lines 146-154 of `DeviceModelTemplate.m` into the `addSpectra` method of `RM3Study.m` (between lines 91 and 92). Then update the error message on line 158 of `DeviceModelTemplate.m` to just say you need to set mu throughout. Some of the tests might need updated to add mu to the input spectra.
1. Maybe update `8spectra.mat` rather than add code to the loading function, just to avoid confusion down the line if someone adds a different `8spectra.mat` and doesn't get that mu is being overwritten.

I'm not totally sold on the second suggestion, but I think the first one would be a useful improvement from this PR, beyond tidying up the example.

ryancoe commented:
I completed 2 with cf7747c and a version of 1 with 705da40.

_**"version of 1:"**_ I moved the checking for `mu` to `WecOptLib.utils.checkSpectrum`, which was already being called in `RM3Study.addSpectra`. I do not check the spectral weighting within `DeviceModelTemplate.m`, because it seems like this would be checking them twice (right?).

Additionally, I added some unit tests for this functionality to `checkSpectrumTest`.

[test_results.pdf](https://github.com/SNL-WaterPower/WecOptTool/files/4595704/test_results.pdf)

H0R5E commented:
I'll try and get the review of this done before our meeting tomorrow.

H0R5E commented on toolbox/+WecOptLib/+utils/checkSpectrum.m:
```diff
@@ -1,100 +1,123 @@
+    end
+end
+
+function [S] = checkWeightingsPresent(S)
```
Consider making this a standalone function, then checkSpectrum doesn't need to return S and it's purpose remains clear. I think the name could be changed as well, to something like "fixWeightings", just to show that it changes the input.

ryancoe commented on toolbox/+WecOptLib/+utils/checkSpectrum.m:
```diff
@@ -1,100 +1,123 @@
 %     You should have received a copy of the GNU General Public License
 %     along with WecOptTool.  If not, see <https://www.gnu.org/licenses/>.
 
 function [] = checkSpectrum(S)
```
agreed, but I actually do prefer that its just one function call to do both

H0R5E replied:
I think I am going to push you on this, but not just because it's nasty to have one function with two purposes (ick). I'm most worried that, as we change to the new structure, there is a risk that this function ends up back inside the optimisation loop without the spectra being corrected first, rending the original purpose of stopping the warning showing up in every iteration moot.

If we have an explicit function that fixes the mu prior to use in any optimisation that the user creates, then checkSpectrum can simply generate an error if it's not set (for more than one spectrum), which is it's typical behaviour, allowing it to be safely used anywhere in the code. Does that make sense?

ryancoe commented:
Based on #143 and #144, we can close this by simply updating the .mat files for the example spectra. Also the warnings would happened only once (outside of loop), so not a big deal anyways.

H0R5E commented:
.mat files added to #161
